torch/_subclasses/fake_tensor.py

```diff
@@ -132,17 +143,32 @@ def check_non_fake_tensor(x):
                 torch.ops.aten._to_copy(input, **new_kwargs), out_device
             )
 
-    if _is_tensor_constructor(func):
+    if _is_tensor_constructor(func) or func in _like_tensor_constructors:
         assert func not in _non_kwarg_device_constructors
         _, new_kwargs = normalize_function(
             func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
         )
-        # cpu is default device if none is specified
-        out_device = new_kwargs.pop("device", torch.device("cpu"))
+        if func in _like_tensor_constructors:
+            default_device = new_kwargs["input"].device
+            # TODO: file issue
+            args = (new_kwargs.pop("input"),)
```

Comment:
This is kind of confusing because you did a normalize to turn everything into kwargs and now you're overwriting args. I did a double take thinking we were missing args but it turns out you're not